File: mobile/lib/shared/models/country.dart
import 'operator.dart';

class Country {
  final String code;
  final String nameFr;
  final String nameEn;
  final String defaultCurrency;
  final List<Operator> operators;

  const Country({
    required this.code,
    required this.nameFr,
    required this.nameEn,
    required this.defaultCurrency,
    required this.operators,
  });

  factory Country.fromJson(Map<String, dynamic> json) {
    return Country(
      code: json['code'] as String,
      nameFr: json['name_fr'] as String? ?? json['name'] as String,
      nameEn: json['name_en'] as String? ?? json['name'] as String,
      defaultCurrency: json['default_currency'] as String? ?? 'XOF',
      operators: (json['operators'] as List<dynamic>?)
              ?.map((o) => Operator.fromJson(o as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() => {
        'code': code,
        'name_fr': nameFr,
        'name_en': nameEn,
        'default_currency': defaultCurrency,
        'operators': operators.map((o) => o.toJson()).toList(),
      };

  @override
  String toString() => 'Country($code, $nameFr)';
}

File: backend/app/models/country.py
from __future__ import annotations

from datetime import datetime, timezone
from typing import List, TYPE_CHECKING

from sqlalchemy import Boolean, DateTime, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base

if TYPE_CHECKING:
    from app.models.operator import Operator


class Country(Base):
    __tablename__ = "countries"

    code: Mapped[str] = mapped_column(String(3), primary_key=True)  # e.g. "BF"
    name_fr: Mapped[str] = mapped_column(String(128), nullable=False)
    name_en: Mapped[str] = mapped_column(String(128), nullable=False)
    default_currency: Mapped[str] = mapped_column(String(8), nullable=False, default="XOF")
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        server_default=func.now(),
    )

    # Relationships
    operators: Mapped[List["Operator"]] = relationship(
        "Operator", back_populates="country", cascade="all, delete-orphan"
    )
